### PR TITLE
EFCore database with repository pattern

### PR DESCRIPTION
**Pull request by SkarpKat**
# TLDR
Made new DB with EFCore, made it follow Reposetory pattern and intialize the data from the new file given from Helge.
Tests are not refactord
## Points to touch up on
The package and the way we initialize

# Indepth
Okay the EFCore Database has been made and the migration files aswell. As of now all the functions work however there may be some functionality with the page and amount of cheeps. The Repo pattern has also been added so the front end interacts through a repository. The DBInitializtion file has been added and it works however the way i made it initialize probably needs to be revisited. There may also be a package which can be removed. Test needs to be refactored to new DB System.
billede

# Update
### 09-10-2023 ~16:00
I have now added the 32 pr page with the new database
### 09-10-2023 ~20:00
The repo has been changed from static to non-static
There has been made a function which should allow for custom made databses
### 09-10-2023 ~22:30
Added the orderd by decending function to get the latest cheeps
Added methods to the cheep reposetory so you can add cheeps to the database and your own temp database for testing

# Me
tumblr_oa6btpvA0p1sx92edo1_400

### CODE REVIEW

File: src/Chirp.Razor/Program.cs [hunk 1]
```diff
@@ -1,7 +1,11 @@
 using Chirp.Razor.Pages;
+using CheepDB;
+using Repository;
 
 var builder = WebApplication.CreateBuilder(args);
 
+CheepRepository.InitDB();
+
```
Comment (Lukan707) [inline src/Chirp.Razor/Program.cs:8]: Just curious. What are the arguments for why it should by changed? :)

**Review by Lukan707 — CHANGES_REQUESTED** [review]
Great job. It looks neat, and reads well. Awesome.
I have a few questions, to specific changes in the code, which I have posted above.

Also, I think we are missing som code, which orders the cheeps in the database by publication date (timestamp) in descending order. Please note that this need to be done, before why offset and limit the result set (selects to only retrieve 32 cheeps for the current page).
I could however be wrong, and this is already working. In that case, please enlighting me on that, as I seem to have missed it.

Overall, it looks really sensible. :)

File: .github/workflows/main_bdsagroup04chirpremotedb.yml [hunk 2]
```diff
@@ -23,10 +23,10 @@ jobs:
           include-prerelease: true
 
       - name: Build with dotnet
-        run: dotnet build ./src/Chirp.Razor/ --configuration Release
+        run: dotnet build ./src/Chirp.CSVDBService/ --configuration Release
```
[the remark this answers is not in the document]
Comment (SkarpKat) [reply .github/workflows/main_bdsagroup04chirpremotedb.yml:26]: This is in a sense is not suposed to exist anymore since we don't use chirpremotedb

File: src/Chirp.Razor/CheepRepository.cs [hunk 3]
```diff
@@ -3,19 +3,22 @@
...
+
+namespace Repository;
+
+public static class CheepRepository
 {
-    private ChirpDBContext db;
+    private static readonly ChirpDBContext db;
 
-    public CheepRepository()
+    static CheepRepository()
```
[the remark this answers is not in the document]
Comment (SkarpKat) [reply src/Chirp.Razor/CheepRepository.cs:13]: The reason it was static is that i forgot to change it back. I used static to test the methods without the need to make a class since i was not sure where to put it i will change this back so it's easier to test with. Thanks for catching this :D

File: src/Chirp.Razor/Program.cs (hunk 1 above)
Comment (SkarpKat) [reply src/Chirp.Razor/Program.cs:8]: The reaseon is that it's static and it's not supposed to be. When it gets changed back to a class we need to move it since it probably is not a good idea to have it there. The static method was just to see it work and dump all the data in the database :D

File: src/Chirp.Razor/CheepRepository.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Lukan707) [reply src/Chirp.Razor/CheepRepository.cs:13]: Nice. Thanks for the reply. Will you make a new commit on the branch? It will automatically add the commit to the current pull request.

File: src/Chirp.Razor/Program.cs (hunk 1 above)
Comment (Lukan707) [reply src/Chirp.Razor/Program.cs:8]: Cool. Maybe change already then? if you commit and push to the same branch, it will automacally add it to this pull request.

Comment (SkarpKat) [reply src/Chirp.Razor/Program.cs:8]: It has been changed

**Comment by SkarpKat** [comment]
> Great job. It looks neat, and reads well. Awesome.
> 
> I have a few questions, to specific changes in the code, which I have posted below.
> 
> 
> 
> Also, I think we are missing som code, which orders the cheeps in the database by publication date (timestamp) in descending order. Please note that this need to be done, before why offset and limit the result set (selects to only retrieve 32 cheeps for the current page).
> 
> I could however be wrong, and this is already working. In that case, please enlighting me on that, as I seem to have missed it.
> 
> 
> 
> Overall, it looks really sensible. :)

I will add the order by later and push it here. As for the 32 per page solution, this was made after the first iteration of getting 32 cheeps so I refactored it to work here as well.

File: src/Chirp.Razor/Program.cs (hunk 1 above)
Comment (Lukan707) [reply src/Chirp.Razor/Program.cs:8]: Nice. Great

**Comment by Lukan707** [comment]
Nice. Tell me, if you want me to do anything.

**Review by Lukan707 — APPROVED** [review]

**Comment by Lukan707** [comment]
Cool. Really nice work!

**Review by AlbinoLoaf — APPROVED** [review]
